lua_modules/cohelper/cohelper.lua

```diff
@@ -0,0 +1,27 @@
+--[[  A coroutine Helper   T. Ellison,  June 2019
+
+This version of couroutine helper demonstrates the use of corouting within
+NodeMCU execution to split structured Lua code into smaller tasks
+
+]]
+--luacheck: read globals node
+
+local modname = ...
+
+local function taskYieldFactory(co)
+  local post = node.task.post
+  return function(nCBs)  -- upval: co,post
+    post(function () -- upval: co, nCBs
+      coroutine.resume(co, nCBs or 0)
+    end)
+    return coroutine.yield() + 1
```

Comment:
I think this should be
```lua
  return function(arg)  -- upval: co,post
    post(function () -- upval: co, arg
      coroutine.resume(co, arg)
    end)
    return coroutine.yield()
```
so as to not be specific to passing around numbers. (i.e., renamed `nCBs` to `arg` and removed the `or 0` and `+ 1`)

Reply:
Try running it :smile:  You approach works fine, but there are circumstances when you _will_ wantto pass context between the two threads to keep synchronisation.